### PR TITLE
dev/emc: Release FV3 solver updates

### PR DESCRIPTION
**Pull request by laurenchilutti**
**Description**

Cherry-picks changes form PR #261 to dev/emc 
This includes  updates on behalf of Lucas Harris including:
Revision to semi-implicit solver to linearize vertical sound wave propagation about the hydrostatic state. This has been extensively tested by GFDL and EMC in SHiELD, GFS, and RRFS.

Fixes # (issue)

**How Has This Been Tested?**

This has been tested with ShiELD in the main branch.

**Checklist:**

Please check all whether they apply or not
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by lharris4** [comment]
Hi, Kate. I believe that this revision to the Rayleigh damping on w is
considerably more sophisticated and effective than that used in the
explicit Rayleigh damping; Kevin V was working on something similar.

If tau_w is disabled in dev/emc is w still damped? This is the behavior in
main and dev/gfdl. Also is anyone using tau_w in dev/emc?

Thanks,
Lucas

**Comment by XiaqiongZhou-NOAA** [comment]
Lucas:
In the current dev/emc, if tau_w=0, w is damped as tau. The purpose of tau_w is to have different damping time scale for horizontal and vertical winds. I am not sure who is currently using this option. This option was added based on Henry's test for WAM at that time. 
Kate

**Comment by XiaqiongZhou-NOAA** [comment]
Lucas:
That sounds a good solution.

Lauren:
Could you revise the code as Lucas suggested?
Thanks.
Kate

**Comment by laurenchilutti** [comment]
I have just pushed the changes, let me know if that is acceptable

**Review by XiaqiongZhou-NOAA — APPROVED** [review]

**Comment by lharris4** [comment]
Hi, Lauren. These changes look good to me. Thanks for taking care of this.

Lucas

File: model/nh_core.F90 [hunk 1]
```diff
@@ -206,15 +206,15 @@ subroutine Riem_Solver3(ms, dt,   is,   ie,   js, je, km, ng,    &
...
+                            pm2, pem, w2, dz2, pt(is:ie,j,1:km), ws(is,j), p_fac, tau_w)
       else
            call SIM_solver(dt, is, ie, km, rdgas, gama, gm2, cp2, akap, &
 #ifdef MULTI_GASES
                            kapad2,  &
 #endif
                            pe2, dm,  &
                            pm2, pem, w2, dz2, pt(is:ie,j,1:km), ws(is,j), &
-                           a_imp, p_fac, scale_m)
+                           a_imp, p_fac, scale_m, tau_w)
```
Comment (XiaqiongZhou-NOAA) [inline model/nh_core.F90:217]: tau_w should be fast_tau_w_sec added in nh_core.F90, nh_utils.F90 ?

Comment (lharris4) [reply model/nh_core.F90:217]: Hi, Kate. I believe that inside the nh_core/nh_utils modules the name is `tau_w` and the namelist flag `fast_tau_w_sec` is used in the call to the NH utils. So this implementation is correct, but potentially confusing. However for consistency's sake and to eliminate confusion we could replace this with `fast_tau_w_sec`.

Thanks,
Lucas

Comment (laurenchilutti) [reply model/nh_core.F90:217]: I will update this also to use fast_tau_w_sec as well. You are right that this could be confusing to follow.

**Comment by laurenchilutti** [comment]
Comments from PR #261 and this PR have all now been addressed.  Changes are identical in both PRs.

**Comment by lharris4** [comment]
Hi, Lauren. This is really great. Thank you so much.

Lucas

**Comment by laurenchilutti** [comment]
replacing this PR with #266

**Pull request closed without merging**